spec.md

```diff
@@ -1190,9 +1191,13 @@ $ curl http://username:password@service-broker-url/v2/service_instances/:instanc
 | 400 Bad Request | MUST be returned if the request is malformed or missing mandatory data. |
 | 422 Unprocessable entity | MUST be returned if the requested change is not supported or if the request cannot currently be fulfilled due to the state of the Service Instance (e.g. Service Instance utilization is over the quota of the requested plan). Additionally, a `422 Unprocessable Entity` MUST be returned if the Service Broker only supports asynchronous update for the requested plan and the request did not include `?accepts_incomplete=true`; in this case the response body MUST contain a error code `"AsyncRequired"` (see [Service Broker Errors](#service-broker-errors)). The error response MAY include a helpful error message in the `description` field such as `"This Service Plan requires client support for asynchronous service operations."`. |
 
-Responses with any other status code MUST be interpreted as a failure.
 When the response includes a 4xx status code, the Service Broker MUST NOT
-apply any of the requested changes to the Service Instance.
+apply any of the requested changes to the Service Instance and the
+Service Instance MUST be in an unmodified and usable state.
+
+Responses with any other status code MUST be interpreted as a failure.
+The Service Instance MUST be considered corrupt and the Platform SHOULD NOT
```

Comment:
I think we should have a section on corrupt instances that explains:

- what the meaning of 'corrupt' is
- when you should set `instance_corrupt` as a broker author
- whether corrupt instances can be updated

Reply:
Agreed.

Reply:
This "Responses with any other status code..." is kind of worrisome. If it said "Any other 5xx error..." then I agree. But what about some 3xx status code? If people agree I can open a new PR to address this since this isn't really part of this PR

Reply:
@fmui ^^

Reply:
actually, does "any other" mean non-4xx or does it mean "not mentioned above in the table or previous paragraph" ?  I think we need to clarify this.

Reply:
I also worry that if a Service Broker has an intermittent failure and returns 500s for a period of time then instances that receive update calls during this time will be marked as failed, leaving them unusable. 

I would prefer the broker be explicit about when an instance is corrupt rather than inferring it from a large range of status codes.